src/cleanlab_codex/project.py

```diff
@@ -145,36 +150,58 @@ def create_access_key(
+        which should include the latest user query and any preceding system or assistant messages.
+
+        **Note:** For single-turn queries, this can be a minimal list with one user message. For multi-turn conversations, provide the full dialog
+        leading up to the final response.
+
+        The function assesses the trustworthiness and quality of the AI `response` in light of the provided `context` and
+        `query`, which should align with the most recent user message in `messages`. If the response is flagged as problematic,
+        Codex is used to retrieve an alternate expert answer or log the query for future SME review.
 
         Args:
-            context (str): The context used by the AI system to generate a response for the query.
-            prompt (str): The full prompt (including system instructions, context, and the original query) used by the AI system to generate a response for the query.
-            query (str): The original user input to the AI system.
-            response (str): The response generated by the AI system for the query.
-            custom_metadata (object, optional): Custom metadata to log in Codex for the query.
-            eval_scores (Dict[str, float], optional): Optional scores to use for the query. When provided, Codex will skip running TrustworthyRAG evaluations on the query and use the provided scores instead.
-            eval_thresholds (Dict[str, float], optional): Optional thresholds to use for evaluating the query. We recommend configuring thresholds on the Project instead and using the same thresholds for all queries.
+            messages (list[ChatCompletionMessageParam]): The full message history from the AI conversation, formatted for OpenAI-style chat completion.
+                This must include the final user message that triggered the AI response. All other arguments—`query`, `context`, and `response`—
+                must correspond specifically to this final user message.
+            response (ChatCompletion | str): The AI-generated response that immediately follows the last user message in `messages`. This is the response being evaluated.
+            context (str): The retrieved context (e.g., from your RAG system) that was supplied to the AI when generating the `response` to the final user query in `messages`.
+            query (str): The original user query that prompted the `response`. This must be the query included in the final user message in `messages`.
+            rewritten_question (str, optional): An optional reformulation of the `query` to make it more self-contained to improve retrieval quality. If this is provided, the query re-writting step is skipped, and the provided `rewritten_question` is used instead.
+            custom_metadata (object, optional): Arbitrary metadata to associate with this validation for logging or analysis inside the Codex project.
```

Comment:
```suggestion
            custom_metadata (object, optional): Arbitrary metadata to associate with this AI response/interaction, that will be logged in the web interface.
```

Reply:
why is this `custom_metadata`, instead of just `metadata` ?

Reply:
backend takes in parameter "custom_metadata" for user metadata, updated to make it clearer on user side to metadata